infra/src/main/kotlin/fi/fta/geoviite/infra/projektivelho/PVClientConfiguration.kt

```diff
@@ -10,17 +9,21 @@ import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.http.HttpHeaders.CONTENT_TYPE
-import org.springframework.http.MediaType.*
-import org.springframework.http.client.reactive.*
+import org.springframework.http.MediaType.APPLICATION_FORM_URLENCODED_VALUE
+import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
 import org.springframework.web.reactive.function.client.ClientRequest
 import org.springframework.web.reactive.function.client.ExchangeFilterFunction
 import org.springframework.web.reactive.function.client.WebClient
 import reactor.core.publisher.Mono
 import reactor.netty.http.client.HttpClient
+import java.time.Duration
 
 val defaultResponseTimeout: Duration = Duration.ofMinutes(5L)
 
-const val MAX_FILE_SIZE: Int = 100 * 1024 * 1024
+const val BYTES_PER_MEGABYTE: Int = 1024 * 1024
+const val MAX_FILE_SIZE: Int = 100 * BYTES_PER_MEGABYTE
+const val MAX_FILE_BUFFER_SIZE: Int = MAX_FILE_SIZE + 1 * BYTES_PER_MEGABYTE
```

Comment:
Miksikäs tähän lisätään megatavu? 🤔 Onko ideana että tasan 100MB tiedostot eivät mahtuisi puskuriin syystä tai toisesta?

Reply:
Joo juurikin se että jostain teknisestä syystä projektivelhon metadatassa raportoitu koko ei jostain syystä olisikan ihan tismalleen sama kuin mitä se on kun se luetaan puskuriin. En tiedä onko tuo oikea huoli mutta tuli vain mieleen ja on aika ärsyttävä testatakaan (pitäisi saada projektivelhoon tasan 100MB tiedosto ladattavaksi). Tuo logiikka siis nyt jättää lataamatta jos velho sanoo sen tiedostokoon olevan yli MAX_FILE_SIZE ja puskurin kooksi otetaan sitten se + vähän just-in-case.